packages/odyssey-react-mui/src/labs/UiShell/UiShellContent.tsx

```diff
@@ -120,6 +134,14 @@ const UiShellContent = ({
       </StyledBannersContainer>
 
       <StyledSideNavContainer>
+        {
+          /* If SideNav should be initially visible and we have not yet received props, render SideNav with minimal inputs */
+          initiallyVisible?.includes(SideNavComponent) && !sideNavProps && (
+            <ErrorBoundary fallback={null} onError={onError}>
+              <SideNav isLoading appName="" sideNavItems={[]} />
+            </ErrorBoundary>
+          )
+        }
```

Comment:
I'm fine with this, except `sideNavItems={[]}` needs to have a statically-defined array, or it can render more than it should. We can add the compiler to React 18 through a 3rd party library.

Might be worth adding here to avoid unnecessary code since this runs in its own sandbox. I wonder how that affects Odyssey or if Odyssey needs to compile with the compiler as well.

I guess it's easier to fix this and worry about the compiler in React v18 after React v19 releases.

Reply:
This is lacking in documentation:
![image](https://github.com/user-attachments/assets/7f17ea9a-8850-4d2e-9dde-2e88cedf0aa3)

I think you need the Babel version (`babel-plugin-react-compiler`) even with Vite. I would just hard-code memoization for now.

Reply:
Another option, make `sideNavItems` optional if `isLoading: true`. That would also solve it, but in that component, you can't do `sideNavItems = []`, it'd still need to be static or memoized somewhere.